libflux/src/core/ast/mod.rs

```diff
@@ -531,7 +531,24 @@ pub struct TypeExpression {
     pub base: BaseNode,
     pub monotype: MonoType,
 }
-
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct Record {
+    #[serde(skip_serializing_if = "BaseNode::is_empty")]
+    #[serde(default)]
+    #[serde(flatten)]
+    pub base: BaseNode,
+    pub tvar: Option<MonoType>,
```

Comment:
If we're checking to make sure this is a type variable instead of just an identifier, then this should probably be

    pub tvar: Option<TvarType>,